infra/resources/_modules/storage_audit/storage_audit.tf

```diff
@@ -0,0 +1,108 @@
+###
+# Immutable Audit Log Storage
+###
+module "immutable_cdc_audit_logs_storage" {
+  source  = "pagopa-dx/azure-storage-account/azurerm"
+  version = "~> 1.0"
+
+  environment = {
+    prefix          = var.prefix
+    env_short       = var.env_short
+    location        = var.location
+    domain          = var.domain
+    app_name        = var.app_name
+    instance_number = var.instance_number
+  }
+
+  resource_group_name = var.resource_group_name
+  subnet_pep_id       = var.subnet_pep_id
+
+  blob_features = {
+    versioning          = true
+    restore_policy_days = 0
+    immutability_policy = {
+      enabled                       = true
+      allow_protected_append_writes = false
+      period_since_creation_in_days = var.cdc_storage_immutability_policy_days
+    }
+  }
+
+  action_group_id = var.action_group_id
+
+  tier = "l"
+
+  tags = var.tags
+}
+
+
+module "immutable_cdc_audit_logs_storage_customer_managed_key" {
+  source               = "git::https://github.com/pagopa/terraform-azurerm-v4//storage_account_customer_managed_key?ref=v7.26.3"
+  tenant_id            = var.tenant_id
+  location             = var.location
+  resource_group_name  = var.resource_group_name
+  key_vault_id         = var.key_vault_id
+  key_name             = format("%s-key", module.immutable_cdc_audit_logs_storage.name)
+  storage_id           = module.immutable_cdc_audit_logs_storage.id
+  storage_principal_id = module.immutable_cdc_audit_logs_storage.principal_id
+}
+
+resource "azurerm_private_endpoint" "immutable_cdc_audit_logs_storage_blob" {
```

Comment:
The private endpoint is managed by Storage Account DX module: https://registry.terraform.io/modules/pagopa-dx/azure-storage-account/azurerm/latest